app/src/main/java/org/oppia/android/app/survey/SurveyWelcomeDialogFragment.kt

```diff
@@ -38,21 +38,22 @@ class SurveyWelcomeDialogFragment : InjectableDialogFragment() {
       explorationId: String,
       mandatoryQuestionNames: List<SurveyQuestionName>
     ): SurveyWelcomeDialogFragment {
+      val args = SurveyWelcomeDialogFragmentArguments.newBuilder().apply {
+        this.topicId = topicId
+        this.explorationId = explorationId
+        this.addAllQuestionNo(
+          SurveyWelcomeDialogFragment().extractQuestions(mandatoryQuestionNames).asList()
```

Comment:
Since this is being called within the same class, I think the Qualifier is redudant.
```suggestion
          extractQuestions(mandatoryQuestionNames).asList()
```

Reply:
The `extractQuestions` method cannot be called without creating an instance.